### ISSUE TITLE
Get images from camera (1 inch)

### ISSUE THREAD

**Issue by ricardosutana** [Insta360Develop/CameraSDK-Cpp#33]
Hi I'm working on camera sdk to get the fisheye images.
I was able to compile and test the sdk functionalities.
Looking into function onVideoData we have
`void OnVideoData(const uint8_t* data, size_t size, int64_t timestamp, uint8_t streamType, int stream_index = 0)`
I suppouse the images are encoded on format h264 and are stored on `data` pointer.
How can I decode this image and display it in live stream mode?

**Comment by bouviervj** [comment]
Yes so this is a h264 stream - and in some configuration you'll get 2 streams (stream_index) - i.e. 3072x3072 configurations - 2 streams corresponding to each sensor - you can take any library to decode the stream, I used ffmpeg to do it - taken from ffmpeg examples - I had to find the right decoder and ffmpeg library to get nvidia decoder 'h264_cuvid' - but I started by the standard decoder, it worked to but consumes a lot of CPU - [precompiled ffmpeg](https://www.gyan.dev/ffmpeg/builds/#release-builds) : follow this example - [decode_video.c](https://ffmpeg.org/doxygen/trunk/decode_video_8c-example.html) - instead of reading from the file , feed the stream we got from insta libraries - once decoded you can inject the image in whichever framework you need to do processing - note: if you take the result back and forth from the GPU to the CPU it might not be optimal .

**Comment by andreamassetti** [comment]
Hi @bouviervj, your answer is very interesting, have you experienced any problems with artifacts and dropped frames when reading the stream with ffmpeg?
For now I'm converting the raw 01.h264 file to mp4 using ffmpeg with the following command: 
`ffmpeg -framerate 24 -i ./01.h264 -c copy ./output.mp4`
But I see a lot of artifacts and when there is any sort of movement and the video hangs quite often for a few frames.

**Comment by bouviervj** [comment]
It's been a while - I  just rebooted the project - what I remember is sometimes a sensor blackout - I suppose it was the temperature of the device. I'm using the 1-inch camera and it get hot. As soon as I have the stream and the video parameters I could have the full sensor view. I did see some artifacts/ non alignment on non-compatible setup - i.e. the camera frame was not centered - or sometimes the video feed was incorrect/corrupted. Apart from that I have really pure reception of the data that I sent to my GPU from processing/stitching.
